include/aos/common/thread.hpp

```diff
@@ -11,13 +11,14 @@
 #include <pthread.h>
 
 #include "aos/common/error.hpp"
+#include "aos/common/noncopyable.hpp"
 
 namespace aos {
 
 /**
  * Aos thread.
  */
-class Thread {
+class Thread : private NonCopyable {
```

Comment:
I suggest adding a move constructor and move assignment operator to this class because the delete constructor has suppressed the default implementation of these operations.

Reply:
We will not use move semantics for now.